web/client/components/map/enhancers/withScalesDenominators.js

```diff
@@ -0,0 +1,43 @@
+
+/*
+ * Copyright 2021, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { getResolutionsForScales } from '../../../utils/MapUtils';
+
+const withScalesDenominators = (Component) => {
```

Comment:
We need a bit of jsdoc for this component, you could take this as an example

https://github.com/geosolutions-it/MapStore2/blob/0625df45f2aca341c79d91711f263c5744bb7eb2/web/client/components/misc/enhancers/localizedProps.js